### ISSUE TITLE
Read Stream of Regions from Stdin or Unix Socket

### ISSUE THREAD

**Issue by Ferdi265** [Ferdi265/wl-mirror#4]
As an extension to #2 (Mirror a region of the screen), wl-mirror could support a mode where it reads a stream of regions from Stdin or a Unix domain socket, to be able to update the selected region while the program is running.

One use case for this could be a presentation setup where the presenter could zoom into an area of the screen to highlight some detail and afterwards zoom back out.

Pros and Cons of input sources
- Stdin: simple and easy, but more of a hassle to set up if you want to combine it with keyboard shortcuts to change region
- Unix Socket: a bit more complicated to set up, but something like `wl-mirror-ctl set-region $REGION` could be easily done from anywhere

**Comment by Ferdi265** [comment]
Work on this is progressing nicely. I still need to iron out a few bugs, but this will likely release before the end of the year.

I went with the stdin approach, since it adds less complexity to `wl-mirror` (although string maniplulation is always complex in C).

For controlling `wl-mirror`'s stdin from keyboard shortcuts or scripts I came up with [pipectl](https://github.com/Ferdi265/pipectl), a small tool that manages short-lived named pipes.
